### PR TITLE
Display targets areas in the map

### PR DESCRIPTION
**Pull request by RobertoCarrilloAvila**
## Screenshots:
Screenshot 2024-02-02 at 12 58 22 a m

### CODE REVIEW

File: src/components/Home/Home.js [hunk 1]
```diff
@@ -19,7 +20,9 @@ const Home = () => {
   }[displayedComponent];
 
   return (
-    <MapContext.Provider value={{ selectedLocation, setSelectedLocation }}>
+    <MapContext.Provider
+      value={{ mapProperties, setMapProperties, targets, setTargets }}
+    >
```
Comment (afmicc) [inline src/components/Home/Home.js:25]: you could define a custom provider to encapsulate all the state logic related to this context. check this [article](https://devtrium.com/posts/how-use-react-context-pro#extract-the-react-context-logic-in-another-file)

File: src/components/CreateTarget/CreateTarget.js [hunk 2]
```diff
@@ -89,7 +94,7 @@ const CreateTarget = ({ onContinue }) => {
           <button
             type="submit"
             className="create-target__submit btn"
-            disabled={!radius || !title || !topic}
+            disabled={!mapProperties.SelectedRadius || !title || !topic}
```
Comment (afmicc) [inline src/components/CreateTarget/CreateTarget.js:97]: is this ok? 

```suggestion
            disabled={!mapProperties.selectedRadius || !title || !topic}
```

File: src/components/CreateTarget/CreateTarget.js [hunk 3]
```diff
@@ -10,10 +10,9 @@ import target from 'assets/icons/target.svg';
 import 'components/CreateTarget/CreateTarget.scss';
 
 const CreateTarget = ({ onContinue }) => {
-  const { selectedLocation } = useContext(MapContext);
+  const { mapProperties, setMapProperties } = useContext(MapContext);
```
Comment (afmicc) [inline src/components/CreateTarget/CreateTarget.js:13]: you could destructure this too 
```suggestion
  const { mapProperties: { selectedRadius, ... }, setMapProperties } = useContext(MapContext);
```

File: src/components/Map/Map.js [hunk 4]
```diff
@@ -1,33 +1,52 @@
...
+  const { mapProperties, setMapProperties, targets, setTargets } =
+    useContext(MapContext);
   const { isLoaded, loadError } = useLoadScript({
     googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAPS_API_KEY,
   });
 
   useEffect(() => {
+    fetchCurrentLocation();
+    fetchTargets();
+  }, []);
```
Comment (afmicc) [inline src/components/Map/Map.js:26]: image

File: src/components/Map/Map.js [hunk 5]
```diff
@@ -1,33 +1,52 @@
...
     navigator.geolocation.getCurrentPosition((position) => {
-      setCurrentLocation({
-        lat: position.coords.latitude,
-        lng: position.coords.longitude,
+      setMapProperties({
+        ...mapProperties,
+        location: {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude,
+        },
```
Comment (afmicc) [inline src/components/Map/Map.js:35]: on your custom context hook, you can simplify this state management avoiding consumer caring about the current state

Comment (flomosq) [reply src/components/Map/Map.js:35]: +1

File: src/components/Map/Map.js [hunk 6]
```diff
@@ -43,15 +62,55 @@ const Map = ({ onSelectLocation }) => {
...
+              />
+
+              <Circle
+                key={`circle-${id}`}
+                center={{ lat: latitude, lng: longitude }}
+                radius={radius}
+                options={MapConfig.targetsOptions}
+              />
+            </>
+          )
```
Comment (flomosq) [inline src/components/Map/Map.js:113]: I usually prefer to separate this into a separate component, `Targets` and render that component here. Looks cleaner IMO

File: src/components/Target/Target.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+      <Circle
+        key={`circle-${id}`}
+        center={{ lat: latitude, lng: longitude }}
+        radius={radius}
+        options={MapConfig.targetsOptions}
+      />
+    </>
+  );
+};
```
Comment (flomosq) [inline src/components/Target/Target.js:25]:
```suggestion
const Target = ({ id, latitude, longitude, radius, icon }) => (
  <>
    <Marker
      key={`marker-${id}`}
      position={{ lat: latitude, lng: longitude }}
      icon={{
        url: icon,
        scaledSize: new window.google.maps.Size(40, 40),
      }}
    />

    <Circle
      key={`circle-${id}`}
      center={{ lat: latitude, lng: longitude }}
      radius={radius}
      options={MapConfig.targetsOptions}
    />
  </>
);
```

File: src/components/Map/Map.js [hunk 8]
```diff
@@ -43,15 +63,46 @@ const Map = ({ onSelectLocation }) => {
...
+          }) => (
+            <Target
+              key={id}
+              id={id}
+              latitude={latitude}
+              longitude={longitude}
+              radius={radius}
+              icon={icon}
+            />
+          )
```
Comment (flomosq) [inline src/components/Map/Map.js:105]:
```suggestion
        {targets.map(({ ...target, topic: { icon } }) => (
            <Target
              {...target}
              icon={icon}
            />
          )}
```

File: src/components/Home/Home.js (hunk 1 above)
Comment (RobertoCarrilloAvila) [reply src/components/Home/Home.js]: Done!

File: src/contexts/MapContext.js [hunk 9]
```diff
@@ -1,5 +1,43 @@
-import { createContext } from 'react';
+import { createContext, useState, useEffect } from 'react';
+import TargetsService from 'services/TargetsService';
 
 const MapContext = createContext();
 
-export default MapContext;
+const MapContextProvider = ({ children }) => {
```
Comment (afmicc) [inline src/contexts/MapContext.js:6]: nice 😄

**Review by afmicc — APPROVED** [review]

**Review by bmarcio20 — APPROVED** [review]
LGTM! I left non-blocking comments.

File: src/components/Map/Map.js [hunk 10]
```diff
@@ -43,15 +39,36 @@ const Map = ({ onSelectLocation }) => {
...
+        )}
+
+        {targets.map(
+          ({
+            target,
+            target: {
+              topic: { icon },
+            },
+          }) => (
+            <Target key={target.id} {...target} icon={icon} />
```
[the remark this answers is not in the document]
Comment (RobertoCarrilloAvila) [reply src/components/Map/Map.js:70]: I had it that way before 😂

File: src/components/CreateTarget/CreateTarget.js [hunk 11]
```diff
@@ -10,10 +10,12 @@ import target from 'assets/icons/target.svg';
 import 'components/CreateTarget/CreateTarget.scss';
 
 const CreateTarget = ({ onContinue }) => {
-  const { selectedLocation } = useContext(MapContext);
+  const {
+    mapProperties: { selectedRadius, selectedLocation },
+    setMapProperties,
+  } = useContext(MapContext);
```
[the remark this answers is not in the document]
Comment (RobertoCarrilloAvila) [reply src/components/CreateTarget/CreateTarget.js]: Done!

**Pull request merged**